File: 10. Prefix Sums/560. Subarray Sum Equals K/Solution.py
from typing import List

class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = 0
        prefix_sum = 0
        freq_map = {0:1}

        for num in nums:
            prefix_sum += num
            if prefix_sum - k in freq_map:
                res += freq_map[prefix_sum - k]
            freq_map[prefix_sum] = freq_map.get(prefix_sum, 0) + 1
        
        return res

File: 10. Prefix Sums/560. Subarray Sum Equals K/Solution.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    int subarraySum(std::vector<int>& nums, int k) {
        int res = 0;
        int prefixSum = 0;
        std::unordered_map<int, int> freqMap = {{0,1}};

        for (const int& num : nums)
        {
            prefixSum += num;

            const auto& it = freqMap.find(prefixSum - k);
            if (it != freqMap.end())
            {
                res += it->second;
            }

            ++freqMap[prefixSum];
        }

        return res;
    }
};